app/javascript/mastodon/actions/store.js

```diff
@@ -1,11 +1,11 @@
 import Immutable from 'immutable';
 
 export const STORE_HYDRATE = 'STORE_HYDRATE';
-
+const NumberIsNaN = (number) => typeof number === 'number' && isNaN(number);
 const convertState = rawState =>
   Immutable.fromJS(rawState, (k, v) =>
     Immutable.Iterable.isIndexed(v) ? v.toList() : v.toMap().mapKeys(x =>
-      Number.isNaN(x * 1) ? x : x * 1));
+      NumberIsNaN(x * 1) ? x : x * 1));
```

Comment:
`babel-polyfill` is included [`Number.isNaN`](https://github.com/zloirock/core-js/blob/v2.4.1/modules/es6.number.is-nan.js).